_posts/2021-06-26-complete-guide-to-cors.md

```diff
@@ -0,0 +1,441 @@
+---
+title: "Preliminary Guide to CORS"
+categories: [craft]
+date: 2021-06-14 06:00:00 +1000
+modified: 2021-06-13 06:00:00 +1000
+author: pratikdas
+excerpt: "In this article, we will understand cross-origin resource sharing (CORS), different types of CORS requests, CORS headers, and describe some common examples of security vulnerabilities caused by CORS misconfigurations along with best practices for secure CORS implementations."
+image:
+  auto: 0074-stack
+---
+
+“CORS” stands for Cross-Origin Resource Sharing that defines the protocol to use between a web browser and a server to determine whether a cross-origin request is allowed. 
+
+In this article, we will understand the following aspects about CORS:
+- Cross-origin resource sharing (CORS) standard
+- Different types of CORS requests
+- CORS headers, and 
+- Some examples of security vulnerabilities caused by CORS misconfigurations 
+- Best practices for secure CORS implementations
+
+{% include github-project.html url="https://github.com/thombergs/code-examples/tree/master/cors" %}
+
+## What is CORS
+CORS is a security standard implemented by browsers that enable scripts running in browsers to access resources located outside of the browser's domain. The CORS policy is published under the [Fetch standard](https://fetch.spec.whatwg.org) defined by the [WHATWG](https://whatwg.org) community which also publishes many web standards like [HTML5](https://html.spec.whatwg.org/multipage/), [DOM](https://dom.spec.whatwg.org), and [URL](https://url.spec.whatwg.org).
+
```

Comment:
A diagram would help tremendously here to make clear what CORS is. It would also help to understand the rest of the article much easier. 

Maybe a sequence diagram with the Browser, "Origin Server", and "Cross-Origin Server" as in my doodle here: 
![20210709_080031](https://user-images.githubusercontent.com/1363578/124996202-e612cd80-e08b-11eb-8677-5d3aca8a6a45.jpg)

Then, use the terms "Origin server" and "Cross-Origin Server" throughout the rest of the article consistently to make everything very clear.